src/lib/Zikula/Core/Helper/ExtensionVar.php

```diff
@@ -0,0 +1,283 @@
+<?php
+/**
+ * Copyright 2015 Zikula Foundation
+ *
+ * This work is contributed to the Zikula Foundation under one or more
+ * Contributor Agreements and licensed to You under the following license:
+ *
+ * @license GNU/LGPLv3 (or at your option, any later version).
+ * @package Zikula
+ *
+ * Please see the NOTICE file distributed with this source code for further
+ * information regarding copyright and licensing.
+ */
+
+namespace Zikula\Core\Helper;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Zikula\Core\Doctrine\Entity\ExtensionVarEntity;
+
+/**
+ * Class ExtensionVar
+ * @package Zikula\Core\Helper
+ *
+ * This class manages the storage and retrieval of extension variables and is the intended replacement
+ * for ModUtil::* methods (getVar, setVar, etc) as well as similar functionality in System:: and ThemeUtil::
+ */
+class ExtensionVar
+{
+    const CONFIG_MODULE = 'ZConfig';
+
+    /**
+     * @var EntityManagerInterface
+     */
+    private $entityManager;
+    /**
+     * @var KernelInterface
+     */
+    private $kernel;
+
+    /**
+     * @var \ArrayObject
+     * @todo consider \Symfony\Component\DependencyInjection\ParameterBag\ParameterBag
+     * @todo consider \Symfony\Component\HttpFoundation\ParameterBag
+     */
+    private $variables;
+
+    /**
+     * ExtensionVar constructor.
+     * @param EntityManagerInterface $entityManager
+     * @param KernelInterface $kernel
+     */
+    public function __construct(EntityManagerInterface $entityManager, KernelInterface $kernel)
+    {
+        $this->entityManager = $entityManager;
+        $this->kernel = $kernel;
+        $this->initExtensionVars();
```

Comment:
I don't like this. IMHO the constructor shouldn't do heavy things like loading vars from the DB. The better solution would be to add an $initialized flag and load the vars in all the methods below.